man/man1/p11sak.1.in

```diff
@@ -1,678 +1,1759 @@
-.B p11sak can be used to generate, list and delete the token keys in an openCryptoki token repository.
-The utility provides a flexible key management tool in openCryptoki to list and generate symmetric (DES, 3DES, AES, AES-XTS) and asymmetric (RSA, EC, IBM Dilithium, IBM Kyber) keys.
-This tool is especially capable of a well defined listing of keys with their PKCS #11 attributes.
-.
+.B p11sak
+.BR \-\-version | \-v
 .
 .
-.SH COMMANDS
-The \fBp11sak\fP tool can operate in three modes: when command
-.I generate-key
-is specified, it operates in the mode to generate a token key in the openCryptoki token repository.
-If command
-.I list-key
-is given, it lists the keys specified in the arguments.
-If command
-.I remove-key
-is given, it removes the keys specified in the arguments.
 .
+.SH DESCRIPTION
+The \fBp11sak\fP tool can be used to manage token keys in a PKCS\~#11 token
+repository. The utility provides a flexible key management tool to generate,
+list, remove, update, copy, import, and export symmetric (DES, 3DES, generic,
+AES, AES\-XTS) and asymmetric (RSA, DH, DSA, EC, IBM Dilithium, IBM Kyber) keys.
+This tool is also capable of listing the keys with their PKCS\~#11 attributes
+and its values (not all attributes may be displayed if a keys is sensitive).
 .PP
-.SS "generate-key"
-.PP
-Use the
-.B generate-key|gen-key|gen
-command and key argument to generate a token key with the respective
-.RI [ ARGS ]
-and
-.RB [ OPTIONS ].
+.B Note:
 The
-.BR \-\-help | \-h
-option will show the arguments and options available.
+.B p11sak
+tool only operates on token keys (i.e. \fBCKA_TOKEN=TRUE\fP), but not on
+session keys (\fBCKA_TOKEN=FALSE\fP). Token keys are stored persistently in the
+token's repository, while session keys are not stored persistently, and only
+exist as long as the session is alive. Thus, session keys generated or imported
+by the
+.B p11sak
+tool would not exist anymore when the
+.B p11sak
+tool has exited.
 .
-.PP
-.SS "list-key"
-.PP
-Use the
-.B list-key|ls-key|ls
-command and key argument to list token keys given the respective
-.RI [ ARGS ]
-and
-.RB [ OPTIONS ].
-The
-.BR \-\-help | \-h
-option will show the arguments and options available.
 .
+.
+.SH COMMANDS
+The \fBp11sak\fP tool supports various commands to generate, list, remove,
+update, import, and export token keys in a PKCS\~#11 token repository.
 .PP
-.SS "remove-key"
+.SS "Generating symmetric and asymmetric keys"
+.B p11sak
+.BR generate\-key | gen\-key | gen
+.I KEYTYPE
+.RB [ ARGUMENTS ]
+.RB [ OPTIONS ]
 .PP
 Use the
-.B remove-key|rm-key|rm
-command and key argument to delete token keys given the respective
-.RI [ ARGS ]
+.BR generate\-key | gen\-key | gen
+command to generate a token key of the specified
+.I KEYTYPE
+with the respective
+.B ARGUMENTS
 and
-.RB [ OPTIONS ].
-The
+.BR OPTIONS .
+Possible values for the
+.I KEYTYPE
+argument are:
+.BR des | 3des | generic | aes | aes\-xts | rsa | dh | dsa | ec | \
+ibm\-dilithium | ibm-kyber .
```

Comment:
ibm\-kyber ?  And remove the space before the dot?

Reply:
Changed to `ibm\-kyber `.
The blank before the dot is correct, the line uses `.BR` formatting, so it toggles between bold and regular. Thus the key types are bold, but the `|` and the ending dot is regular. There will be no blank between those when viewing the man page.